doppler/resource_secret.go

```diff
@@ -0,0 +1,122 @@
+package doppler
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceSecret() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSecretUpdate,
+		ReadContext:   resourceSecretRead,
+		UpdateContext: resourceSecretUpdate,
+		DeleteContext: resourceSecretDelete,
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"config": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"value": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"computed": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceSecretUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(APIClient)
+
+	var diags diag.Diagnostics
+	project := d.Get("project").(string)
+	config := d.Get("config").(string)
+	name := d.Get("name").(string)
+	value := d.Get("value").(string)
+	if value != "" {
+		newSecret := RawSecret{Name: name, Value: &value}
+		err := client.UpdateSecrets(ctx, project, config, []RawSecret{newSecret})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	d.SetId(getSecretId(project, config, name))
+
+	resourceSecretRead(ctx, d, m)
```

Comment:
Should we do something with this function's return value?

Reply:
Great catch! Terraform's example project doesn't even handle this